### ISSUE TITLE
Compound assignment of delegates do not report nullability mismatch warnings

### ISSUE THREAD

**Issue by RikkiGibson** [dotnet/roslyn#85708]
```cs
using System;

class Program
{
    public event Action<string?>? E;

    static void Main()
    {
        var p = new Program();
        p.EWrapper += s => s.ToString();
        p.E?.Invoke(null); // No warning, but an NRE is reported at runtime

        // A similar case involving combining delegates produces warnings,
        // but only when compound assignment is not used
        Action<string> a1 = s => s.ToString();
        Action<string?> a2 = s => { };
        a2 = a1 + a2; // warning
        a2 += a1; // no warning
        a2(null);
    }

    public event Action<string>? EWrapper
    {
        add => E += value; // Probably, a nullable warning should be reported here, whether 'E' field is accessible or not
        remove => E -= value;
    }
}
```

**Comment by MihuBot** [comment]
I'm a bot. Here is a possible related and/or duplicate issue (I may be wrong):
- https://github.com/dotnet/roslyn/issues/31018

**Comment by RikkiGibson** [comment]
MihuBot did much better than an advance check with Copilot.
